src/campaign.cairo

```diff
@@ -106,9 +106,10 @@ mod TokengiverCampaign {
             salt: felt252,
             recipient: ContractAddress
         ) -> ContractAddress {
+            let caller = get_caller_address();
             let count: u16 = self.count.read() + 1;
             let token_giverNft_contract_address = self
-                .deploy_token_giver_nft(self.token_giver_nft_class_hash.read(), count.into());
+                .deploy_token_giver_nft(self.token_giver_nft_class_hash.read(), caller);
```

Comment:
caller is the owner of the campaign, 

Good call.

Reply:
> Good call.

Thanks.